app/controllers/ems_cluster_controller.rb

```diff
@@ -122,12 +122,17 @@ def set_config(db_record)
   end
 
   def breadcrumbs_options
+    breadcrumbs = [
+      {:title => _("Compute")},
+      {:title => _("Infrastructure")},
+      {:title => _("Clusters / Deployment Roles")}
+    ]
+    header_title = title_for_clusters
+    breadcrumbs.push(:title => header_title) unless header_title == _("Clusters / Deployment Roles")
```

Comment:
My point here is that we should use more robust mechanism than comparing one gettext string to another: should the behavior of `header_title()` change, your code will stop working.

Also, why are we even adding another breadcrumb here? Why can't the BC structure here be:
```
Compute > Infrastructure > Clusters / Deployment Roles
```

and for particular cluster:

```
Compute > Infrastructure > Clusters / Deployment Roles > Particular Cluster
```

Reply:
The header is part of the breadcrumbs and the header is changing according to the user content.

```ruby
case EmsCluster.node_types
when :non_openstack
  _("Clusters")
when :openstack
  _("Deployment Roles")
else
  _("Clusters / Deployment Roles")
end
``` 
Ex. if user has `:non_openstack` == `node_types`:
**BC**
`Compute > Infrastructure > Clusters / Deployment Roles > Clusters`

**Particular cluster**
`Compute > Infrastructure > Clusters / Deployment Roles > Particular Cluster`

And the `Clusters` part is missing, but it should not, because it is part of a hierarchy and user is going deeper:

`Compute > Infrastructure > Clusters / Deployment Roles > Clusters > Particular Cluster`

Reply:
Right, I understand that part. But I think that if `Clusters / Deployment Roles` is already part of the breadcrumb, then we don't need `Clusters` in it, do we?